### PR TITLE
Support reading host schemas

### LINKED ISSUE #3: Doesn't see host schemas
Since dconf-editor doesn't have access to host schemas, it (for the most part) only shows changed keys.

### PR DESCRIPTION
**Pull request by refi64**
Closes #3. Pretty simple overall, I just update XDG_DATA_DIRS in the Flatpak, but use symlinks to avoid polluting it with other host stuff we may not actually want.

### CODE REVIEW

**Comment by refi64** [comment]
Ugh so dconf's appdata file failed validation, I tried temporarily disabling it for now.

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-appstream-check": true
```
Comment (TingPing) [inline flathub.json]: Fix this in a different PR.

File: start-dconf-editor.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  if [[ "$dir" == /usr/* ]]; then
+    dir=/run/host/"$dir"
+  fi
+
+  schemas="$dir/glib-2.0/schemas"
+  if [[ -d "$schemas" ]]; then
+    bridged=$(mktemp -d XXXXXXXXXX -p $bridge_dir)
+    mkdir -p $bridged/glib-2.0
+    ln -s "$schemas" $bridged/glib-2.0
+    XDG_DATA_DIRS=$XDG_DATA_DIRS:$bridged
```
Comment (TingPing) [inline start-dconf-editor.sh]: Honestly I am not a fan of this. While it is certainly the most low effort solution `XDG_DATA_DIRS` is full of stuff we absolutely do not want to read.

I believe what would be safer (I've not tested it) is for you to bring everything into a dir, run `glib-compile-schemas` and then set `GSETTINGS_SCHEMA_DIR`.

Comment (TingPing) [reply start-dconf-editor.sh]: I should probably read the actual script better. I see this limits it to the `glib-2.0` directory which should be good.

**Comment by refi64** [comment]
@TingPing Removed the flathub.json for now.

**Comment by Exalm** [comment]
Interesting. What happens if the same schema is present both on host and in sandbox, but is different?

**Comment by TingPing** [comment]
> Interesting. What happens if the same schema is present both on host and in sandbox, but is different?

First one wins I believe (https://github.com/GNOME/glib/blob/3e7baaae237f480c07ae2e52831ea10db60dac47/gio/gsettingsschema.c#L350-L355). If there is an conflicting dconf/gtk one then potentially gsettings will just fatally error.

**Comment by Erick555** [comment]
```
$ shellcheck start-dconf-editor.sh

In start-dconf-editor.sh line 3:
IFS=: read -ra host_data_dirs < <(flatpak-spawn --host sh -c 'echo $XDG_DATA_DIRS')
                                                             ^-------------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.


In start-dconf-editor.sh line 8:
mkdir -p $bridge_dir
         ^---------^ SC2086: Double quote to prevent globbing and word splitting.


In start-dconf-editor.sh line 17:
    bridged=$(mktemp -d XXXXXXXXXX -p $bridge_dir)
                                      ^---------^ SC2086: Double quote to prevent globbing and word splitting.


In start-dconf-editor.sh line 18:
    mkdir -p $bridged/glib-2.0
             ^------^ SC2086: Double quote to prevent globbing and word splitting.


In start-dconf-editor.sh line 19:
    ln -s "$schemas" $bridged/glib-2.0
                     ^------^ SC2086: Double quote to prevent globbing and word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2016 -- Expressions don't expand in singl...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...

```

**Comment by ramcq** [comment]
FWIW, I tested and this works. Very pleased! :)

**Comment by refi64** [comment]
@Erick555 technically bridge_dir won't ever have spaces in it, but I can quote it for consistency. The first warning is incorrect however; the unexpanded string is intentionally passed so it can be expanded on the host end.

**Comment by refi64** [comment]
Also, I don't see host & sandbox conflicts being a huge issue; if the schemas are 100% incompatible, then we'd be in huge trouble as for the dconf database anyway. The host will most likely only ever have an older version, which would probably be compatible, so I don't see anything too crazy potentially happening.

If someone on GNOME 3.30 wants to try it out though to make sure there's no compat issues, be my guest!

**Comment by Erick555** [comment]
>  technically bridge_dir won't ever have spaces in it, but I can quote it for consistency. The first warning is incorrect however; the unexpanded string is intentionally passed so it can be expanded on the host end.

Ok, please leave the latter alone and quote the former, thx.

**Comment by refi64** [comment]
@Erick555 Done!

**Pull request merged**